### PR TITLE
Add features to CSV export

### PR DESCRIPTION
**Pull request by Tianchi-Liu**
- warn and name unnamed DataArrays so that they can be exported
- rename columns to ease column access in R (no spaces / special characters, start with a letter or a dot, unique)
- add DataArray variable unit to header if it exists as the `units` attribute
- warn about Excel column limit in addition to row limit

To test: export data in notebooks with export sections.

### CODE REVIEW

**Review by nicolejkeeney — CHANGES_REQUESTED** [review]
I'm unable to unzip the csv file generated by the export function once I download it from the hub to my local drive. 

cannot_unzip

File: climakitae/data_export.py [hunk 1]
```diff
@@ -30,27 +28,91 @@ def _export_to_netcdf(data_to_export, save_name, **kwargs):
...
+    None
+
     """
+    if not data_to_export.name:
+        warnings.warn(
+            (
+                "The unnamed dataset cannot be converted to CSV. "
+                "It is being named 'data'."
+            )
+        )
```
[the remark this answers is not in the document]
Comment (Tianchi-Liu) [reply climakitae/data_export.py:87]: The Archive Utility error could be due to interruption during file download. Could you try downloading again or unzipping through the Terminal? I was able to unzip on Windows and don't have a Mac to test on.

[the remark this answers is not in the document]
Comment (elehmer) [reply climakitae/data_export.py:87]: Yeah I agree, just go ahead and say no filename prefix was given so naming it 'data'.

**Review by elehmer — COMMENTED** [review]
Looks good. Just the minor comments

**Review by nicolejkeeney — APPROVED** [review]
You're right that I can unzip the file using the command line. I think we should think about guidance for our users on how to do so. I don't love that Mac users will encounter this error and be required to understand how to use their command line to access downloaded data. But that isn't related to this PR I'm realizing and could be addressed in a separate PR, so I've approved this one :D

File: climakitae/data_export.py [hunk 2]
```diff
@@ -30,27 +28,91 @@ def _export_to_netcdf(data_to_export, save_name, **kwargs):
...
+        data table to update
+    variable : string
+        name of the variable column
+    unit : string
+        unit associated with the variable
+
+    Returns
+    -------
+    df : pandas.DataFrame
+        data table with the variable unit added to its header
```
[the remark this answers is not in the document]
Comment (Tianchi-Liu) [reply climakitae/data_export.py]: Great catch. Don't trust code editor defaults!

File: climakitae/data_export.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Tianchi-Liu) [reply climakitae/data_export.py]: Yeah I see how the warning can be confusing. Replaced with a comment on why naming is necessary.

**Pull request merged**